logrotateandupload.sh

```diff
@@ -0,0 +1,213 @@
+#!/bin/bash
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright 2019 Joyent, Inc.
+#
+
+#
+# Rotate log files using logadm and push /var/log/manta/upload/... log files up
+# to Manta.
+#
+
+echo ""   # blank line in log file helps scroll btwn instances
+set -o errexit
+set -o pipefail
+export PS4='[\D{%FT%TZ}] ${BASHPID}: ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+set -o xtrace
+
+
+## Environment setup
+
+export PATH=/opt/local/bin:$PATH
+
+## Global variables
+
+# Immutables
+
+SSH_KEY=/root/.ssh/id_rsa
+LOCKFILE=/tmp/log_upload_lockfile
```

Comment:
Speaking of the `finish` function it seems like due to [this](https://github.com/joyent/manta-scripts/blob/MANTA-1936/logrotateandupload.sh#L158) line that none of the stale PID stuff is actually going to work as expected and it seems like that line makes [this](https://github.com/joyent/manta-scripts/blob/MANTA-1936/logrotateandupload.sh#L213-L215) redundant.

Reply:
Er, I might misunderstand what the "EXIT" `trap` does, but my read of that is:

- if we exit, remove the TEMPFILE in case we crashed in the middle of `function create_lockfile`

I'm not entirely sure without playing, however. I haven't combed carefully through `create_lockfile`.

Reply:
I caught my mistake. I realized this was for TEMPFILE and not the regular LOCKFILE so it makes sense to me now.